cpp/perspective/src/cpp/gnode.cpp

```diff
@@ -597,10 +646,16 @@ t_gnode::get_table_sptr() {
 void
 t_gnode::promote_column(const std::string& name, t_dtype new_type) {
     PSP_TRACE_SENTINEL();
-    PSP_VERBOSE_ASSERT(m_init, "touching uninited object");
+    PSP_VERBOSE_ASSERT(m_init, "Cannot `promote_column` on an uninited gnode.");
```

Comment:
Much better error message!  I assume you changed these to trace the segfault?  Reminds me that a future improvement may be to completely replace `PSP_VERBOSE_ASSERT` which check object initialization only and replace with proper valgrind support in the build scripts, stricter clang warnings and our existing excellent test coverage?